File: app/src/main/java/net/emilla/struct/sort/IndexWindow.java
package net.emilla.struct.sort;

public final class IndexWindow {

    public final int start;
    public final int last;
    public final int end;

    /*internal*/ IndexWindow(int start, int last) {
        this.start = start;
        this.last = last;
        this.end = last + 1;
    }

    public int size() {
        return end - start;
    }
}

File: app/src/main/java/net/emilla/struct/sort/IndexWindow.kt
package net.emilla.struct.sort

@ConsistentCopyVisibility
data class IndexWindow internal constructor(
    @JvmField val start: Int,
    @JvmField val last: Int
): IndexPortion {

    @JvmField
    val end = last + 1
    @JvmField
    val size = end - start

    fun isEmpty() = start == end
    fun arrayIndex(windowIndex: Int) = start + windowIndex
}
